packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -2,55 +2,159 @@
+import { gql } from '@apollo/client';
+import { useNavigate } from '@reach/router';
+import { thaw } from 'icepick';
 import { useBooleanState } from 'fxa-react/lib/hooks';
-import { gql, useMutation } from '@apollo/client';
+import { useHandledMutation } from '../../lib/hooks';
+import { useAccount, Email, Account, useVerifiedAction } from '../../models';
 import UnitRow from '../UnitRow';
-import AlertBar from '../AlertBar';
-import { useAccount, Email } from '../../models';
-import sentryMetrics from 'fxa-shared/lib/sentry';
-import { Link, useNavigate } from '@reach/router';
+import AlertBar, { AlertBarType } from '../AlertBar';
+import ModalVerifySession from '../ModalVerifySession';
+import { ReactComponent as TrashIcon } from './trash-icon.svg';
 
-export const RESEND_SECONDARY_EMAIL_CODE_MUTATION = gql`
+export const RESEND_EMAIL_CODE_MUTATION = gql`
   mutation resendSecondaryEmailCode($input: EmailInput!) {
     resendSecondaryEmailCode(input: $input) {
       clientMutationId
     }
   }
 `;
 
+export const MAKE_EMAIL_PRIMARY_MUTATION = gql`
+  mutation updatePrimaryEmail($input: EmailInput!) {
+    updatePrimaryEmail(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const DELETE_EMAIL_MUTATION = gql`
+  mutation deleteSecondaryEmail($input: EmailInput!) {
+    deleteSecondaryEmail(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
 type UnitRowSecondaryEmailContentAndActionsProps = {
   secondary: Email;
   isLastVerifiedSecondaryEmail: boolean;
 };
 
 export const UnitRowSecondaryEmail = () => {
   const account = useAccount();
+  const navigate = useNavigate();
+  const [verifiedAction, setVerifiedAction, verified] = useVerifiedAction();
   const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
   const [email, setEmail] = useState<string>();
-  const navigate = useNavigate();
+  const [alertMessage, setAlertMessage] = useState<[string, AlertBarType]>();
+
   const secondaryEmails = account.emails.filter((email) => !email.isPrimary);
   const hasAtLeastOneSecondaryEmail = !!secondaryEmails.length;
   const lastVerifiedSecondaryEmailIndex = secondaryEmails
     .map((email) => email.verified)
     .lastIndexOf(true);
 
-  // TODO: DRY this up and don't import `sentryMetrics` into every component
-  // needing a mutation - we must have an `onError` option in mutations
-  // to allow tests to pass but providing one prevents an error from actually
-  // throwing so we need to manually report it to Sentry.
-  // See https://github.com/apollographql/react-apollo/issues/2614
-  const [resendSecondaryEmailCode, { error }] = useMutation(
-    RESEND_SECONDARY_EMAIL_CODE_MUTATION,
-    {
-      onError: (error) => {
-        sentryMetrics.captureException(error);
-        revealAlertBar();
-      },
-      onCompleted: () => {
-        navigate('/beta/settings/emails/verify', { state: { email } });
-      },
-    }
+  const showSuccess = useCallback(
+    (message: string) => {
```

Comment:
You're probably right -- it might also be worth baking this right into a hook that also encapsulates the `const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();` stuff. 🤔 Can this be a followup?

Reply:
Yeah, WFM 👍 Will you file it? Let's maybe mark it as higher priority than the `useMutation` convention ticket, in that one I expect we'll go through all of our `useMutations` and make them consistent.

Reply:
Donezo https://jira.mozilla.com/browse/FXA-2152